### DISCUSSION TITLE
Installation of workspace package into `.bin` folder

### DISCUSSION THREAD

**Discussion by woifes** [pnpm/pnpm#5701] [Q&A]
I have a monorepo and wanted to add an helper script as a package of the workspace. I would like to add the package to the devDependency of either the root package.json to have the command accessable in all other projects, or inserting it in the devDepencencies for the packages which shall use the command.
I could no get it working that a local package is linked to the `.bin` folder. The command is also not accessable.

Is this intended?

**Comment by tarasvalstro** [comment]
I have the same question, it doesn't seem to work as expected. I have a bin script on a package in the monorepo, I added it as a dependency in the root package.json but it doesn't get added to `node_modules/.bin`

**Comment by chrisjcthomas** [comment]
I don't think this is intended to fail in the general case.

I just reproduced a minimal workspace locally with:

- a workspace package `tool`
- `"bin": { "hello-tool": "bin/hello.js" }`
- another workspace package depending on `tool` via `workspace:*`

and after `pnpm install`, the bin shim was created both in:

- `node_modules/.bin/hello-tool` at the root when the root depended on it
- `packages/app/node_modules/.bin/hello-tool` when the app package depended on it

So the basic workspace-package-with-bin flow does work.

That suggests the thing to check is not "does pnpm support this at all?" but whether the helper package is shaped like a normal package binary, for example:

- the package has a valid `name`
- the `bin` field points to a real file
- that file exists in the package contents
- the consuming package actually lists it in `dependencies` / `devDependencies`
- you're looking in the consuming package's `node_modules/.bin` (or running it through `pnpm exec`)

So my read is: no, this is not generally unsupported; a workspace package with a valid `bin` entry should be linkable into `.bin`.
